### ISSUE TITLE
Using a Restricted Data Token

### ISSUE THREAD

**Issue by ozzyonfire** [ajmnz/selling-partner-api-node#15]
Thanks for the great package. 

I'm really enjoying the typescript support. However I've run into a couple of issues. 

Is there an option to use a Restricted Data Token? 

I am able to create one. Then I try to pass it in via the `headers` option of the RequestParams, but I have a feeling that the axios request interceptor is overwriting my 'x-amz-access-token' header just before the request is actually called. 

I think we need an option to use a custom x-amz-access-token when we supply a restrictedAccessToken or something. 

I haven't had time to do a pull request, but I can maybe do that in a bit.

**Comment by ozzyonfire** [comment]
An example of what I am trying to do:

```ts
const response = await sp.tokens.createRestrictedDataToken({
  restrictedResources: [{
    method: 'GET',
    path: '/orders/v0/orders',
    dataElements: ['buyerInfo', 'shippingAddress']
  }]
});

const token = response.restrictedAccessToken;

const orders = await sp.orders.getOrders({
  // @ts-ignore
  MarketplaceIds: 'ATVPDKIKX0DER',
  CreatedAfter: '2021-01-01',
  // @ts-ignore
  OrderStatuses: 'Unshipped',
}, {
  headers: {
    'x-amz-access-token': token
  }
});
```

**Comment by ajmnz** [comment]
This is now available in version [2.1.2](https://github.com/ajmnz/selling-partner-api-node/releases/tag/2.1.2)
